### PR TITLE
[ZH] Update `Beatmap\Difficulty`

### PR DESCRIPTION
**Pull request by YumeMuzi**
## Self-check

- [x] The changes are tested against the [contribution checklist](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#self-check)
- [x] *(translations only)* The changes are reviewed on GitHub [by a fluent speaker](https://osu.ppy.sh/wiki/osu!_wiki/Contribution_guide#review)

### CODE REVIEW

File: wiki/Beatmap/Difficulty/zh.md [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
 # 难度
 
-*请勿与 [谱面](/wiki/Beatmap) 相混淆。*
+*请勿与[谱面](/wiki/Beatmap)相混淆。*
 
-一个谱面的**难度**反映了完成它所需的能力水平。难度名称与星级评定通常会给出关于该谱面难度的足够信息。
+**难度**是以[`.osu` 后缀名](/wiki/osu!_File_Formats/Osu_(file_format))存储，用于区分一张谱面内不同的游戏难度等级的文本文件。它包含了为歌曲量身定制、测定好[时间轴](/wiki/Beatmapping/Timing_section)并用于特定[游戏模式](/wiki/Game_mode)的[击打物件](/wiki/Hit_object)。不同难度有不同的定位，对玩家的考验也不同。通常难度名和[谱面星级](/wiki/Beatmapping/Star_rating)就能提示玩家它的游玩难度有多高。
 
-## 难度级别
+难度内的击打物件在不同的游戏模式内可以组成不同样式的[排列](/wiki/Beatmap/Pattern)，它能反映难度的整体外观和给玩家的感觉。谱师可以通过调整繁多的[谱面难度设定](/wiki/Client/Beatmap_editor/Song_Setup#difficulty)或者玩家可以开启不同的[游戏模组 (Mods)](/wiki/Game_modifier) 来影响谱面的实际游玩难度。
```
Comment (kj415j45) [inline wiki/Beatmap/Difficulty/zh.md:20]:
```suggestion
各模式下不同的击打物件所构成的[排列](/wiki/Beatmap/Pattern)，为难度的观感奠定基调。谱师可以通过调整多样的[谱面难度设定](/wiki/Client/Beatmap_editor/Song_Setup#difficulty)或者玩家可以开启不同的[游戏模组（Mods）](/wiki/Game_modifier) 来影响谱面的实际游玩难度。
```
4个不能算繁多吧 :thinking:

第一句还需要再斟酌一下，我这个译法肯定也不行。

File: wiki/Beatmap/Difficulty/zh.md [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
   - stars
-outdated: true
-outdated_since: a1c56faa4ae6d2ba10579e0e2c9971b41fe4fa33
+  - 难度名
+  - 难度星级
+  - 谱面星级
+  - 星数
 ---
 
 # 难度
```
Comment (kj415j45) [inline wiki/Beatmap/Difficulty/zh.md:14]:
```suggestion
# 难度（Difficulty）
```

File: wiki/Beatmap/Difficulty/zh.md [hunk 3]
```diff
@@ -1,25 +1,38 @@
...
 
-*请勿与 [谱面](/wiki/Beatmap) 相混淆。*
+*请勿与[谱面](/wiki/Beatmap)相混淆。*
```
Comment (kj415j45) [inline wiki/Beatmap/Difficulty/zh.md:16]:
```suggestion
*请勿与[谱面（Beatmap）](/wiki/Beatmap)相混淆。*
```

File: wiki/Beatmap/Difficulty/zh.md [hunk 4]
```diff
@@ -1,25 +1,38 @@
...
 
-一个谱面的**难度**反映了完成它所需的能力水平。难度名称与星级评定通常会给出关于该谱面难度的足够信息。
+**难度**是以[`.osu` 后缀名](/wiki/osu!_File_Formats/Osu_(file_format))存储，用于区分一张谱面内不同的游戏难度等级的文本文件。它包含了为歌曲量身定制、测定好[时间轴](/wiki/Beatmapping/Timing_section)并用于特定[游戏模式](/wiki/Game_mode)的[击打物件](/wiki/Hit_object)。不同难度有不同的定位，对玩家的考验也不同。通常难度名和[谱面星级](/wiki/Beatmapping/Star_rating)就能提示玩家它的游玩难度有多高。
```
Comment (kj415j45) [inline wiki/Beatmap/Difficulty/zh.md:18]:
```suggestion
**难度**是以[`.osu` 后缀名](/wiki/osu!_File_Formats/Osu_(file_format))存储，用于区分一张谱面内不同的游戏难度等级的文本文件。它包含了为歌曲量身定制、测定好[时间轴](/wiki/Beatmapping/Timing_section)和用于特定[游戏模式](/wiki/Game_mode)的[击打物件](/wiki/Hit_object)。不同难度有不同的定位，对玩家的考验也不同。通常难度名和[谱面星级](/wiki/Beatmapping/Star_rating)就能提示玩家它的游玩难度有多高。
```

File: wiki/Beatmap/Difficulty/zh.md [hunk 5]
```diff
@@ -57,11 +70,13 @@ outdated_since: a1c56faa4ae6d2ba10579e0e2c9971b41fe4fa33
...
-## 星级评定
+## 谱面星级
+
+自2021年7月27日之后，osu! 官网会自动使用以下光谱方案给谱面赋予一个指示谱面星级的颜色。
 
-![星级评定范围](/wiki/shared/star-rating/spectrum.png)
+![osu! 谱面星级光谱](/wiki/shared/star-rating/spectrum.png)
 
-在官网上，基于星级评定的范围，谱面会被自动给予一个不同的难度图标：
+在 [AIMod](/wiki/Client/Beatmap_editor/AiMod) 工具内或一些场合下，会按照以下谱面星级范围给难度赋予一个特定的标识：
```
Comment (kj415j45) [inline wiki/Beatmap/Difficulty/zh.md:79]:
```suggestion
在 [AiMod](/wiki/Client/Beatmap_editor/AiMod) 工具内或一些场合下，会按照以下谱面星级范围给难度赋予一个特定的标识：
```

File: wiki/Beatmap/Difficulty/zh.md (hunk 1 above)
Comment (YumeMuzi) [reply wiki/Beatmap/Difficulty/zh.md:20]: 差不多拉 我觉得这句没啥问题 keep

**Review by kj415j45 — DISMISSED** [review]

File: wiki/Beatmap/Difficulty/zh.md [hunk 6]
```diff
@@ -1,25 +1,38 @@
...
 ---
 
-# 难度
+# 难度（Difficulty）
 
-*请勿与 [谱面](/wiki/Beatmap) 相混淆。*
+*请勿与[谱面（Beatmap）](/wiki/Beatmap)相混淆。*
 
-一个谱面的**难度**反映了完成它所需的能力水平。难度名称与星级评定通常会给出关于该谱面难度的足够信息。
+**难度**是以[`.osu` 后缀名](/wiki/osu!_File_Formats/Osu_(file_format))存储，用于区分一张谱面内不同的游戏难度等级的文本文件。它包含了为歌曲量身定制、测定好[时间轴](/wiki/Beatmapping/Timing_section)和用于特定[游戏模式](/wiki/Game_mode)的[击打物件](/wiki/Hit_object)。不同难度有不同的定位，对玩家的考验也不同。通常难度名和[谱面星级](/wiki/Beatmapping/Star_rating)就能提示玩家它的游玩难度有多高。
```
Comment (RushFTK) [inline wiki/Beatmap/Difficulty/zh.md:18]:
```suggestion
谱面中的**难度**是一个以[`.osu` 后缀名]文本文件(/wiki/osu!_File_Formats/Osu_(file_format))存储的游戏关卡。它包含了为歌曲量身定制、测定好[时间轴](/wiki/Beatmapping/Timing_section)和用于特定[游戏模式](/wiki/Game_mode)的[击打物件](/wiki/Hit_object)。不同难度有不同的定位，对玩家的考验也不同。通常难度名和[谱面星级](/wiki/Beatmapping/Star_rating)足以显示难度所具有的挑战性。
```
**你必须进行修改**

难度不是一个文件，只是以文件的形式存储。

[关卡的中文维基](https://zh.wikipedia.org/wiki/%E9%97%9C%E5%8D%A1_(%E9%9B%BB%E5%AD%90%E9%81%8A%E6%88%B2))

> A beatmap's difficulty is a game level, which is stored as a text file with [the .osu extension](https://github.com/ppy/osu-wiki/blob/master/wiki/osu!_File_Formats/Osu_(file_format)). It consists of [hit objects](https://github.com/ppy/osu-wiki/blob/master/wiki/Hit_object) of a certain [game mode](https://github.com/ppy/osu-wiki/blob/master/wiki/Game_mode), which are set up accordingly to the [timing](https://github.com/ppy/osu-wiki/blob/master/wiki/Beatmapping/Timing_section) tailored to a specific song.

File: wiki/Beatmap/Difficulty/zh.md [hunk 7]
```diff
@@ -1,25 +1,38 @@
...
 
-## 难度级别
+难度内的击打物件在不同的游戏模式内可以组成不同样式的[排列](/wiki/Beatmap/Pattern)，它能反映难度的整体外观和给玩家的感觉。谱师可以通过调整繁多的[谱面难度设定](/wiki/Client/Beatmap_editor/Song_Setup#difficulty)或者玩家可以开启不同的[游戏模组 (Mods)](/wiki/Game_modifier) 来影响谱面的实际游玩难度。
```
Comment (RushFTK) [inline wiki/Beatmap/Difficulty/zh.md:20]:
```suggestion
据游戏模式、游玩难度、音乐的旋律及给人带来感觉的不同，难度中击打物件所组成的[排列](/wiki/Beatmap/Pattern)不尽相同。谱师可以通过调整多种[谱面难度设定](/wiki/Client/Beatmap_editor/Song_Setup#difficulty)或者玩家可以开启不同的[游戏模组 (Mods)](/wiki/Game_modifier) 来影响谱面的实际游玩难度。
```
"音乐的旋律及给人带来感觉"不知道直接用音调音色合适不合适，感觉你的翻译与原文有些出入。  
我也觉得4种也不叫繁多……主要原文也没有相应的含义。。。 
> A beatmap's difficulty is a game level, which is stored as a text file with [the .osu extension](https://github.com/ppy/osu-wiki/blob/master/wiki/osu!_File_Formats/Osu_(file_format)). It consists of [hit objects](https://github.com/ppy/osu-wiki/blob/master/wiki/Hit_object) of a certain [game mode](https://github.com/ppy/osu-wiki/blob/master/wiki/Game_mode), which are set up accordingly to the [timing](https://github.com/ppy/osu-wiki/blob/master/wiki/Beatmapping/Timing_section) tailored to a specific song. Difficulties have different aspects, which directly affect user experience. The difficulty name and [star rating](https://github.com/ppy/osu-wiki/blob/master/wiki/Beatmapping/Star_rating) usually give enough information to tell how challenging it is.

File: wiki/Beatmap/Difficulty/zh.md [hunk 8]
```diff
@@ -1,25 +1,38 @@
...
-*请参见：[难度的命名](/wiki/Ranking_Criteria/Difficulty_naming)*
+难度可以通过提供一组或多组音频样本（或[音效](/wiki/Beatmapping/Hitsound)）来增强歌曲，提供额外的听觉反馈层次。
 
-以下是对应某个谱面难度的常见难度等级。在各自模式的 [Ranking 准则](/wiki/Ranking_Criteria) 中可以找到它们的具体定义和要求。
+<!-- TODO: this description of visuals needs to be improved:
+  - a separate section;
+  - a very small paragraph on every key component, including those from Beatmap#overview;
+-->
+
+与排列不同的是，难度的[视觉组件](/wiki/Beatmap)通常能提供辅助层次的体验，在大多数情况下，也可以在[视觉设置](/wiki/Client/Interface/Visual_settings)页面关闭它们。某些谱面所含的特殊视觉组件（如：[故事板](/wiki/Storyboard)或者[谱面皮肤](/wiki/Skinning)）通常与谱面组成一个整体，而且大多数视觉组件在不同难度内的显示效果也不同（如：[kiai 时间](/wiki/Gameplay/Kiai_time)和 [combo 颜色](/wiki/Glossary/Combo_colour)）。
```
Comment (RushFTK) [inline wiki/Beatmap/Difficulty/zh.md:29]:
```suggestion
与排列不同的是，难度的[视觉组件](/wiki/Beatmap)通常能提供辅助层次的体验，在大多数情况下，也可以在[视觉设置](/wiki/Client/Interface/Visual_settings)页面关闭它们。大多数视觉组件（如：[kiai 时间](/wiki/Gameplay/Kiai_time)和 [combo 颜色](/wiki/Glossary/Combo_colour)）各难度相互独立，而有些视觉组件（如：[故事板](/wiki/Storyboard)或者[谱面皮肤](/wiki/Skinning)）是谱面所有难度共享的。
```

> While some of them, such as [storyboards](https://github.com/ppy/osu-wiki/blob/master/wiki/Storyboard) or [custom skins](https://github.com/ppy/osu-wiki/blob/master/wiki/Skinning), belong to the beatmap as a whole, most of the visual components may vary per difficulty (e.g., [kiai time](https://github.com/ppy/osu-wiki/blob/master/wiki/Gameplay/Kiai_time) sections and [combo colours](https://github.com/ppy/osu-wiki/blob/master/wiki/Glossary/Combo_colour)).

**Review by RushFTK — CHANGES_REQUESTED** [review]

**Review by RushFTK — APPROVED** [review]

**Review by Walavouchey — APPROVED** [review]

**Pull request merged**